build.sbt

```diff
@@ -161,51 +163,55 @@ val scalacJvmOptions = Def.setting {
-  ),
-  scalacOptions ++= scalacJvmOptions.value,
-  Compile / mainClass := Some("org.scalafmt.cli.Cli"),
-  nativeImageVersion := "22.3.0",
-  nativeImageInstalled := isCI,
-  nativeImageOptions ++= {
-    val isMusl = sys.env.get("NATIVE_IMAGE_MUSL").exists(_.toBoolean)
-    if (isMusl) Seq("--libc=musl") else Nil
-  },
-  nativeImageOptions ++= {
-    val isStatic = sys.env.get("NATIVE_IMAGE_STATIC").exists(_.toBoolean)
-    if (isStatic) Seq("--static") else Nil
-  },
-).dependsOn(coreJVM, dynamic).enablePlugins(NativeImagePlugin)
+lazy val cli = crossProject(JVMPlatform).withoutSuffixFor(JVMPlatform)
+  .in(file("scalafmt-cli")).settings(
+    moduleName := "scalafmt-cli",
+    assembly / mainClass := Some("org.scalafmt.cli.Cli"),
+    assembly / assemblyOption := (assembly / assemblyOption).value
+      .withPrependShellScript(Some(defaultUniversalScript(shebang = false))),
+    assembly / assemblyJarName := "scalafmt.jar",
+    assembly / assemblyMergeStrategy := {
+      case "reflect.properties" => MergeStrategy.first
+      case x =>
+        val oldStrategy = (assembly / assemblyMergeStrategy).value
+        oldStrategy(x)
+    },
+    libraryDependencies ++= Seq(
+      "com.googlecode.java-diff-utils" % "diffutils" % "1.3.0",
+      "com.martiansoftware" % "nailgun-server" % "0.9.1",
+      "com.github.scopt" %% "scopt" % "4.1.0",
+    ),
+    scalacOptions ++= scalacJvmOptions.value,
+    Compile / mainClass := Some("org.scalafmt.cli.Cli"),
+    nativeImageVersion := "22.3.0",
+    nativeImageInstalled := isCI,
+    nativeImageOptions ++= {
+      val isMusl = sys.env.get("NATIVE_IMAGE_MUSL").exists(_.toBoolean)
+      if (isMusl) Seq("--libc=musl") else Nil
+    },
+    nativeImageOptions ++= {
+      val isStatic = sys.env.get("NATIVE_IMAGE_STATIC").exists(_.toBoolean)
+      if (isStatic) Seq("--static") else Nil
+    },
+  ).dependsOn(core, dynamic).enablePlugins(NativeImagePlugin)
 
-lazy val tests = project.in(file("scalafmt-tests")).settings(
-  publish / skip := true,
-  libraryDependencies ++= Seq(
-    // Test dependencies
-    "com.lihaoyi" %% "scalatags" % "0.13.1",
-    scalametaTestkit,
-    munit.value,
-  ),
-  scalacOptions ++= scalacJvmOptions.value,
-  javaOptions += "-Dfile.encoding=UTF8",
-  buildInfoPackage := "org.scalafmt.tests",
-  buildInfoKeys :=
-    Seq[BuildInfoKey]("resourceDirectory" -> (Test / resourceDirectory).value),
-).enablePlugins(BuildInfoPlugin).dependsOn(coreJVM, dynamic, cli)
+lazy val tests = crossProject(JVMPlatform).withoutSuffixFor(JVMPlatform)
+  .in(file("scalafmt-tests")).settings(
+    publish / skip := true,
+    libraryDependencies ++= Seq(
+      // Test dependencies
+      "com.lihaoyi" %% "scalatags" % "0.13.1",
+      scalametaTestkit,
+      munit.value,
+    ),
+    scalacOptions ++= scalacJvmOptions.value,
+    javaOptions += "-Dfile.encoding=UTF8",
+    buildInfoPackage := "org.scalafmt.tests",
+    buildInfoKeys := Seq[BuildInfoKey](
+      "resourceDirectory" ->
```

Comment:
repeating my question from another pr: why is this needed like this?

Reply:
Do you mean that `Test / resourceDirectory` should still work? I think it should work yeah

Reply:
`(Test / resourceDirectory).value` points to `scalafmt-tests/jvm/src/test/resources`, but in this commit we moved the tests from there to `scalafmt-tests/shared/src/test/resources`. This BuildInfo value is used for the DiffTests, and we do not want to keep them in the jvm directory, as later we'll add the Scala Native targets (where the same BuildInfo value changed here will work)

Reply:
Looks like `Test / resourceDirectories` contains all possible directories. We could use that and search for `shared` or make this a list?

Reply:
I tried making it a list but this ended up being a way bigger change, so now we just filter for the shared directory